crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -258,58 +258,59 @@ impl CumulocityConverter {
 
     async fn try_convert_event(
         &mut self,
+        source: &EntityTopicId,
         input: &Message,
+        event_type: &str,
     ) -> Result<Vec<Message>, ConversionError> {
         let mut messages = Vec::new();
-        let mqtt_topic = input.topic.name.clone();
-        let mqtt_payload = input.payload_str().map_err(|e| {
-            ThinEdgeJsonDeserializerError::FailedToParsePayloadToString {
-                topic: mqtt_topic.clone(),
-                error: e.to_string(),
-            }
-        })?;
 
-        let tedge_event = ThinEdgeEvent::try_from(&mqtt_topic, mqtt_payload).map_err(|e| {
-            ThinEdgeJsonDeserializerError::FailedToParseJsonPayload {
-                topic: mqtt_topic,
-                error: e.to_string(),
-                payload: mqtt_payload.chars().take(50).collect(),
-            }
-        })?;
-        let child_id = tedge_event.source.clone();
-        let need_registration = if let Some(child_id) = child_id.clone() {
-            add_external_device_registration_message(child_id, &mut self.children, &mut messages)
-        } else {
-            false
+        let event_type = match event_type.is_empty() {
+            true => DEFAULT_EVENT_TYPE,
```

Comment:
That's new. Earlier this was an error case, reported as a parsing error on `tedge/errors` topic. It was different for `tedge/measurements` as the type wasn't mandatory for it. But for events and alarms, the `type` was mandatory. So I hope this was intentional. @reubenmiller could you confirm?

Reply:
Yes having a default type used for events will useful as it makes it easier to publish an event, and it aligns with the measurement type handling so it makes for a nice and consistent API.

Reply:
And I'm pretty sure no one is actually subscribing to the `tedge/errors` topics, so usually they just wonder why the event is not being published.